kolibri/core/content/utils/content_request.py

```diff
@@ -471,9 +574,21 @@ def process_download_request(download_request):
     Processes a download request
     :type download_request: ContentDownloadRequest
     """
+    peers = []
     # we do not need to filter by version, since content import should work for any
-    peer = _get_preferred_network_location()
-    if not peer:
+    preferred_peer = _get_preferred_network_location(
+        instance_id=download_request.source_instance_id
+    )
+    if preferred_peer:
+        peers.append(preferred_peer)
+
+    # if we had a preferred peer, add a fallback peer
+    if download_request.source_instance_id is not None:
+        fallback_peer = _get_preferred_network_location()
```

Comment:
Shouldn't we be filtering by Kolibri version here?

Reply:
Definitely! :facepalm:

Reply:
Oh wait, no! This is just the file import, not the metadata import, so it should okay to leave unfiltered?

Reply:
Ahhhh, ok!

Reply:
Maybe add a comment for future people thinking they are being helpful like me?

Reply:
Yep, will do!